apps/examples/log_dump/log_dump_main.c

```diff
@@ -42,13 +44,29 @@ int log_dump_main(int argc, char *argv[])
 {
 	int ret = 1;
 	char buf[READ_BUFFER_SIZE];
-	int fd = open("/proc/logsave", O_RDONLY);
+	int fd = open("/proc/logsave", O_RDWR);
```

Comment:
Suggest you to change the flag for opening to O_RDONLY

```
        /* PROCFS is read-only.  Any attempt to open with any kind of write
	 * access is not permitted.
	 *
	 * REVISIT:  Write-able proc files could be quite useful.
	 */
	if (((oflags & O_WRONLY) != 0 || (oflags & O_RDONLY) == 0) && (!filep->f_inode->u.i_mops->write)) {
		fdbg("ERROR: Only O_RDONLY supported\n");
		return -EACCES;
	}
```

O_RDWR = 0x0002
O_RDONLY = 0x0000
O_WRONLY = 0x0001

We have noticed that the flag O_RDWR often works instead of (O_RDONLY | O_WRONLY) but it is not the same.
If we try to fix this access check on the Procfs side in the future, it might cause problems here.
Please use O_RDONLY instead of you are only wanting to read the file.

Reply:
As discussed in the below, we want to write now.